scenarioo-server/src/main/java/org/scenarioo/business/diffViewer/ComparisonExecutor.java

```diff
@@ -60,99 +59,183 @@
+		Future<BuildDiffInfo> futureComparison = asyncComparisonExecutor.submit(new Callable<BuildDiffInfo>() {
 			@Override
 			public BuildDiffInfo call() {
-				return runComparison(baseBranchName, baseBuildName, comparisonConfiguration);
+				return calculateComparison(baseBranchName, baseBuildName, comparisonConfiguration);
 			}
 		});
-	}
 
-	private BuildDiffInfo runComparison(String baseBranchName, String baseBuildName,
-										ComparisonConfiguration comparisonConfiguration) {
-		ThreadLogAppender comparisonLog = null;
 		ComparisonParameters comparisonParameters = new ComparisonParameters(baseBranchName, baseBuildName, comparisonConfiguration,
-			configurationRepository.getConfiguration().getDiffImageAwtColor());;
-		BuildDiffInfo buildDiffInfo = null;
-		ComparisonConfiguration resolvedComparisonConfiguration = null;
+			configurationRepository.getConfiguration().getDiffImageAwtColor());
 
-		try {
-			comparisonLog = registerLogFile(baseBranchName, baseBuildName, comparisonConfiguration);
-			long startTime = System.currentTimeMillis();
+		storeComparisonScheduled(comparisonParameters);
 
-			LOGGER.info("=== START OF BUILD COMPARISON ===");
-			logBaseBuildAndComparisonConfiguration(baseBranchName, baseBuildName, comparisonConfiguration);
+		return futureComparison;
+	}
 
-			resolvedComparisonConfiguration = resolveComparisonConfiguration(comparisonConfiguration, baseBuildName);
+	private BuildDiffInfo calculateComparison(String baseBranchName, String baseBuildName,
+											  ComparisonConfiguration comparisonConfiguration) {
 
-			if (resolvedComparisonConfiguration == null) {
-				LOGGER.warn("No comparison build found for base build: " + baseBranchName + "/"
-					+ baseBuildName + " with defined comparison: " + comparisonConfiguration.getName());
-			} else {
-				comparisonParameters = new ComparisonParameters(baseBranchName, baseBuildName, resolvedComparisonConfiguration,
-					configurationRepository.getConfiguration().getDiffImageAwtColor());
-				storeComparisonInProgress(comparisonParameters);
-				buildDiffInfo = new UseCaseComparator(comparisonParameters).compare();
-				storeComparisonSuccessful(comparisonParameters, buildDiffInfo);
+			BuildDiffInfo buildDiffInfo;
+			ComparisonConfiguration resolvedComparisonConfiguration;
+			ThreadLogAppender comparisonLog = null;
+			ComparisonParameters comparisonParameters = new ComparisonParameters(baseBranchName, baseBuildName, comparisonConfiguration,
+				configurationRepository.getConfiguration().getDiffImageAwtColor());
+			long startTime = System.currentTimeMillis();
+
+			// Precondition check: do nothing if comparison was allready calculated since scheduling it for calculation.
+			try {
+				if (!isComparisonInStateQueuedForComparison(baseBranchName, baseBuildName, comparisonConfiguration.getName())) {
+					// DO NOTHING: The comparison was already processed somehow inbetween (maybe same comparison was scheduled more than once)
```

Comment:
isComparisonInStateQueuedForComparison() can also be false if the BuildDiffInfo was not found at all. So it could also be the case that it was not calculated yet. Should we try to calculate the comparison in this case?

Reply:
NO - this means there is something wrong and the comparison was not properly queued or it might be that somebody canceled the comparison or removed the folder or whatsoever. I think it's good to not do anything in that case. There will be a log statement.